File: litTrack/litTrack.Services/Database/Narudzba.cs
using System;
using System.Collections.Generic;

namespace litTrack.Services.Database;

public partial class Narudzba
{
    public int NarudzbaId { get; set; }

    public int KorisnikId { get; set; }

    public int NacinPlacanjaId { get; set; }

    public string Sifra { get; set; } = null!;

    public DateTime DatumNarudzbe { get; set; }

    public decimal UkupnaCijena { get; set; }

    public string? StateMachine { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime? VrijemeBrisanja { get; set; }

    public virtual Korisnik Korisnik { get; set; } = null!;

    public virtual NacinPlacanja NacinPlacanja { get; set; } = null!;

    public virtual ICollection<StavkaNarudzbe> StavkaNarudzbes { get; set; } = new List<StavkaNarudzbe>();
}

File: litTrack/UI/littrack_desktop/lib/models/narudzba.dart
import 'package:json_annotation/json_annotation.dart';

part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? narudzbaId;
  String sifra;
  DateTime datumNarudzbe;
  double ukupnaCijena;
  String? stateMachine;
  int korisnikId;
  int nacinPlacanjaId;
  String? imePrezime;
  String? nacinPlacanja;
  int? brojStavki;

  Narudzba(
    this.narudzbaId,
    this.sifra,
    this.datumNarudzbe,
    this.ukupnaCijena,
    this.stateMachine,
    this.korisnikId,
    this.nacinPlacanjaId,
    this.imePrezime,
    this.nacinPlacanja,
    this.brojStavki,
  );

  factory Narudzba.fromJson(Map<String, dynamic> json) => _$NarudzbaFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}